### PR TITLE
Catch incorrect MFA device/code used

### PR DESCRIPTION
**Pull request by bekbulatov**
Resolve https://github.com/claranet/ssha/issues/36

### CODE REVIEW

File: ssha/aws.py [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+    def wrapper(f):
+        @wraps(f)
+        def wrapped(*args, **kwargs):
+            tries = attempts
+            while tries:
+                tries -= 1
+                try:
+                    return f(*args, **kwargs)
+                except (ClientError, ParamValidationError) as error:
+                    if attempts > 0:
```
Comment (raymondbutcher) [inline ssha/aws.py]: Should this be `if tries > 0:` ?

File: ssha/aws.py [hunk 2]
```diff
@@ -1,19 +1,43 @@
...
+            while tries:
+                tries -= 1
+                try:
+                    return f(*args, **kwargs)
+                except (ClientError, ParamValidationError) as error:
+                    if tries > 0:
+                        print(error)
+                    else:
+                        raise
+            return f(*args, **kwargs)
```
Comment (raymondbutcher) [inline ssha/aws.py]: I don't think this line will ever run, unless you used `retry(attempts=0)`. You could delete this line and change `while tries:` to `while True:` and I think it'd do the same thing.

Also I'm not a fan of single letter variables. I think `func` is commonly used in decorators, and it's what I would use.

File: ssha/cli.py [hunk 3]
```diff
@@ -43,3 +44,8 @@ def main():
 
     except KeyboardInterrupt:
         pass
+    except Exception as e:
```
Comment (raymondbutcher) [inline ssha/cli.py]: Another single letter variable.

File: ssha/cli.py [hunk 4]
```diff
@@ -43,3 +44,8 @@ def main():
...
+        if args.verbose:
+            print(traceback.format_exc())
```
Comment (raymondbutcher) [inline ssha/cli.py]: Thoughts on printing the exception verses doing `raise`?

If you prefer printing it, then should it also `return 1` here so the program exits with an error code?

**Review by raymondbutcher — CHANGES_REQUESTED** [review]

File: ssha/cli.py (hunk 4 above)
Comment (bekbulatov) [reply ssha/cli.py]: I think it's not user-friendly to crash with exception. Agreed, I should return a non-zero code

Comment (raymondbutcher) [reply ssha/cli.py]: Isn't printing the exception and then exiting with non-zero the same as crashing, as far as a user can see? (I can't remember what `traceback.format_exc()` looks like but I'm thinking it looks the same as what happens when Python crashes with an exception).

**Comment by bekbulatov** [comment]
Thank you for the review @raymondbutcher. I would agree there is no much value to hide a traceback, it only adds a bit complexity.

**Review by raymondbutcher — APPROVED** [review]

**Pull request merged**